service/pokemon_service.go

```diff
@@ -0,0 +1,30 @@
+package service
+
+import (
+	"errors"
+
+	"github.com/ErickRodriguezWize/academy-go-q42021/domain/model"
+)
+
+var (
+	PokemonMissingError = errors.New("Couldn't find pokemon")
+)
+
+//GetPokemonById finds a pokemon inside pokemons array using the ID as filter. 
+func GetPokemonByID(pokemons []model.Pokemon, ID int) (model.Pokemon, error){
+	targetPkm := model.Pokemon{}
+
+	for _,pkm := range pokemons{
+		if pkm.ID == ID {
+			targetPkm = pkm
+			break
+		}
+	}
+
+	if targetPkm.Name == "" {
+		return targetPkm, PokemonMissingError
+	}
+
+	return targetPkm, nil
+
+}
```

Comment:
No need to break, you could return after you find the pokemon.

```suggestion
func GetPokemonByID(pokemons []model.Pokemon, ID int) (model.Pokemon, error){
	targetPkm := model.Pokemon{}

	for _,pkm := range pokemons{
		if pkm.ID == ID {
			return pkm, nil
		}
	}

	if targetPkm.Name == "" {
		return targetPkm, PokemonMissingError
	}

	return targetPkm, nil
}

```